src/utilities/identities/identities.ts

```diff
@@ -385,6 +389,10 @@ async function syncDidStateWithBlockchain(address: string | null | undefined) {
   if (!wasOnChain && isOnChain) {
     await saveIdentity({ ...identity, did: fullDid });
   }
+  // possible in internal version when switching blockchain endpoints
+  if (isInternal && !wasOnChain && !isOnChain) {
+    await saveIdentity({ ...identity, did: `did:kilt:light:00${kiltAddress}` });
```

Comment:
This is so ugly I’d rather not do that at all :shrug:

Reply:
It's ugly, but I do agree with Raphael that it's annoying in the internal version to have your DID break when you change endpoints. Do you have another suggestion?

Reply:
Honestly I got a little problem with sporran making its own inferences about a DID being deactivated. A DID should only be considered deactivated if you resolve it and it says deactivated. We got a blockchain as a single point of truth, sporran shouldn't be keeping state that diverges from that. Very hypothetical point here but keep in mind that governance can pretty much do anything, including making DIDs disappear.

Reply:
In the bug you describe, when you resolve the DID it returns `null`. There would be no metadata with the deactivated flag, because the DID never existed on Spiritnet. My bugfix handles this case.

Reply:
I think discussion of removing Sporran's internal state is far outside the scope of this PR

Reply:
My point is: the code may look clean, but it is hacky, as it jumps to the _DID deactivated_ state without ever getting notice from DID resolution that the DID has indeed been deactivated. It takes a failure to resolve the DID as equivalent to it being deactivated, but it isn't. That's where the unexpected behaviour that we're seeing when switching chains comes from.